drivers/segger/rtt_initialize.c

```diff
@@ -0,0 +1,53 @@
+/****************************************************************************
+ * drivers/segger/rtt_initialize.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <assert.h>
+
+#include <nuttx/config.h>
+#include <nuttx/kmalloc.h>
+#include <nuttx/segger/rtt.h>
+
+#include <SEGGER_RTT.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: rtt_channel_initialize
+ ****************************************************************************/
+
+void rtt_channel_initialize(int channel, FAR char * name,
```

Comment:
why not do the initialize in lib_rtt[out|in]stream_init?

Reply:
Initializing at system startup is a cleaner way

lib_rttoutstream_init may be called multiple times, which leads to the need to determine whether it has been initialized

Moreover, it is possible to write data through rtt_serial, but lib_rttoutstream_init is not called for initialization